### PR TITLE
Issue/14/create model superclass

### PR DESCRIPTION
**Pull request by cavestruz**
We took out parameters, and now only have the Model superclass and some tests are added for the superclass.

### CODE REVIEW

**Comment by matthewkirby** [comment]
It is worth noting, that the unit tests we wrote will currently fail. We require a parameter object to be defined for them to pass. This is going to be done Monday morning and then the tests will pass. We made a test object to test the tests and the tests pass if we have a Parameter object.

If this test is merged into master, we can merge the travis-ci branch and have continuous integration for all PR into master from now on.

**Comment by cristobal-sifon** [comment]
@matthewkirby 

1. OK, I wasn't aware.

2. You're right that setters are what makes the properties special and as far as I can tell without setters it's a matter of taste/good practice. I defined them here so that the __init__ is clean and short. It makes it easy to follow the code and make changes.

3. Huh, good point. Let me think about it.

File: clmm/tests/test_model.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    assert_raises(TypeError, models.Model, lambda x: x, 'r')
+    assert_raises(TypeError, models.Model, 'x*x', ['r'])
+
+    test_model = models.Model(lambda x: x)
+    assert_block(test_model)
+
+    test_model = models.Model(lambda x: x, ['r'])
+    assert_block(test_model)
+
+    # Everything below should fail until Parameter is defined
```
[the remark this answers is not in the document]
Comment (matthewkirby) [reply clmm/tests/test_model.py]: Done

**Comment by matthewkirby** [comment]
Just to have the record in the issue, this was closed to resolve issue #14

**Pull request merged**